packages/flutter_tools/templates/plugin/android-java.tmpl/build.gradle.tmpl

```diff
@@ -12,29 +9,34 @@ buildscript {
     }
 }
 
+plugins {
+    id "com.android.library"
+}
+
+group = "{{androidIdentifier}}"
+version = "1.0"
+
 rootProject.allprojects {
     repositories {
         google()
         mavenCentral()
     }
 }
 
-apply plugin: "com.android.library"
-
 android {
     if (project.android.hasProperty("namespace")) {
         namespace = "{{androidIdentifier}}"
     }
 
-    compileSdk = {{compileSdkVersion}}
+    compileSdk = flutter.compileSdkVersion
```

Comment:
I think you should add a comment whenever you reference a 'flutter.variable':

// We recommend using flutter.compileSdkVersion as it will automatically
// pickup the flutter sdk's preferred version number. You can override this if
// necessary.

Reply:
Hmm this would mean waiting for the next stable, right? Because the packages repo tests require passing on stable and master, and this won't be available on stable

Reply:
Why remove the template changes? We know they are safe and would only impact newly created projects

Reply:
Well because we are not actually sure of the consequences. We are guessing that it is safe. If we use this in our own plugins then we will have a much better chance of detecting if there are weird corner cases. 

The risk is that people make plugins that we later figure out have a corner case and we have no way to contact those plugins. If we delay updating the template the risk is that if there is no unhandled corner case then we lose easier update support for the plugins made between now and when the templates eventually get updated. 

Given that all plugins are in that state now I think that is a good tradeoff. Open to feedback that I am being too cautious.

Reply:
I would add that our guesses around gradle safety are OK and not GREAT

Reply:
Yes what I am suggesting would required us to wait to use it until the rest of this code was in stable.